tests/test_optimizer.py

```diff
@@ -979,7 +979,10 @@ def test_annotate_types(self):
             result = parse_and_optimize(annotate_types, sql, dialect, dialect=dialect)
 
             with self.subTest(title):
-                self.assertEqual(result.type.sql(), exp.DataType.build(expected).sql())
+                self.assertEqual(
```

Comment:
I noticed this was missing dialect arguments. This now matches `test_annotate_funcs()` below.